packages/flutter/test/gestures/monodrag_test.dart

```diff
@@ -201,3 +254,15 @@ class MockHitTestTarget implements HitTestTarget {
   @override
   void handleEvent(PointerEvent event, HitTestEntry entry) { }
 }
+
+/// A [PanGestureRecognizer] that tries to beat [VerticalDragGestureRecognizer] in the arena.
+///
+/// Typically, [VerticalDragGestureRecognizer] wins because it has a smaller threshold to
+/// accept the gesture. This recognizer uses the same threshold that [VerticalDragGestureRecognizer]
+/// uses.
+class _EagerPanGestureRecognizer extends PanGestureRecognizer {
+  @override
+  bool hasSufficientGlobalDistanceToAccept(PointerDeviceKind pointerDeviceKind, double? deviceTouchSlop) {
```

Comment:
What about all of the other methods?

Reply:
To make the test simpler I didn't included the other methods. This test doesn't care about the fling gesture, so it wouldn't help here.

Reply:
I think the idea behind this comment was: each method that we make public should be included in the test framework.

Reply:
Understood. I'll update it.

Reply:
Updated. 

For our use-case, `primaryDragAxis` could be kept private, since it has a default implementation and we don't need to change it. However, if someone needs to write a custom `VerticalDragRecognizer` or a `HorizontalDragRecognizer` they will need to override `primaryDragAxis`.

Reply:
I think if only some of the methods are required for your use case, we should only expose those needed, rather than exposing all of them.

https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#lazy-programming

Reply:
Updated. Could you please take a look again @Piinks ?

Reply:
Hopefully it was all right for me to edit the original PR comment; I think that table makes it a bit easier to visually parse.

My guess is that a second approval will be contingent on re-privatizing `getDeltaForDetails` and `getPrimaryValueFromOffset`, since they aren't needed for your use case.

Reply:
I think that if those methods should be re-privatized, a default implementation should be provided. Otherwise, it won't be possible to subclass `DragGestureRecognizer` outside of the flutter package.

Reply:
Good point! I can see that as a reason to keep them as public.

…or, we could pull a ***pro gamer move*** and turn `DragGestureRecognizer` into a `sealed class`. This could help streamline the logic within the framework, and since extending `DragGestureRecognizer` directly would no longer be possible, developers wouldn't waste any time trying to do it and could instead extend its subclasses (or the base `OneSequenceGestureRecognizer` class).

Reply:
Would it be possible to extend a class that extends a sealed class? I've never done that. If it works, it works for me that way, we could extend `PanGestureRecognizer`. I'd like to hear @Piinks opinion on that before making this change.

Reply:
> Would it be possible to extend a class that extends a sealed class?

Yes sir! The main goal of a `sealed` class is to enable exhaustive pattern-matching:

```dart
void foo(DragGestureRecognizer recognizer) {
  switch (recognizer) {
    case HorizontalDragGestureRecognizer():
      // ...
    case VerticalDragGestureRecognizer():
      // ...
    case PanGestureRecognizer():
      // ...
  }
}
```

Fortunately, passing a subtype of `PanGestureRecognizer` doesn't break this behavior!

Reply:
Sounds good, I'll give it a try.